task17/index.js

```diff
@@ -0,0 +1,114 @@
+const URL = "https://mocki.io/v1/af7d1b29-da7d-4eba-977b-cdb5caeb2fff";
+
+const slideShowContainer = document.getElementById("js-slideShowContainer");
+const slideShowWrapper = document.getElementById("js-slideShowWrapper");
+const slideImgList = document.getElementById("js-slideImgList");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  try {
+    const value = await fetchImgData();
+    renderSlideImage(value);
+    renderSlideNumber(value);
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+};
+init();
+
+const renderSlideImage = (value) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImage.src = value[i].img;
+    slideImage.alt = value[i].alt;
+    value[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideImgList.appendChild(slideImgFragment);
+};
+
+const renderSlideNumber = (value) => {
+  const slideNumText = createElementWithClassName("p", "slide-number");
+  const slideLength = value.length;
+  slideNumText.id = "js-slideNumber";
+  slideNumText.textContent = `${findIndexItem()} / ${slideLength}`;
+  slideShowContainer.appendChild(slideNumText);
+};
```

Comment:
IMO
There is a `currentIndexImg`, so it would be better to use that.
```suggestion
const renderSlideNumber = (value) => {
  const slideNumText = createElementWithClassName("p", "slide-number");
  const slideLength = value.length;
  slideNumText.id = "js-slideNumber";
  currentIndexImg = value.findIndex((v) => v.display);
  slideNumText.textContent = `${findIndexItem()} / ${slideLength}`;
  slideShowContainer.appendChild(slideNumText);
};
```

Reply:
Sorry, the argument names, etc. are tentative and not appropriate.

以下の部分は、init()の中でやった方が適切な気がしました。
(renderSlideNumber()はカウンター部分をrenderする役割なので
ここで値の代入を行うのは適切でないと思いました...すみません!)
```js
currentIndexImg = value.findIndex((v) => v.display);
```
下記スレッドに私の考えを示したのでご確認お願いします。
https://github.com/seeeen044/JavaScript_handson/pull/18#discussion_r859328704